pkg/security/security_profile/dump/manager.go

```diff
@@ -637,6 +637,34 @@ func (adm *ActivityDumpManager) StopActivityDump(params *api.ActivityDumpStopPar
 	return &api.ActivityDumpStopMessage{Error: errMsg.Error()}, errMsg
 }
 
+// HasActiveActivityDump returns true if the given event has an active dump
+func (adm *ActivityDumpManager) HasActiveActivityDump(event *model.Event) bool {
+	// ignore events with an error
+	if event.Error != nil {
+		return false
+	}
+
+	// is this event sampled for activity dumps ?
+	if !event.IsActivityDumpSample() {
+		return false
+	}
+
+	adm.Lock()
+	defer adm.Unlock()
+
+	for _, d := range adm.activeDumps {
+		d.Lock()
+		matches := d.MatchesSelector(event.ProcessCacheEntry)
+		state := d.state
+		d.Unlock()
+		if matches && state == Running {
+			return true
+		}
+	}
```

Comment:
I wonder if both checks are needed. Is "IsActivityDumpSample" not enough to determine that the event was tagged kernel side for an active dump ?

Reply:
Yes only checking whether the event is an activity dump sample was my initial though as well. But I decided to reproduce the checks that are made the `ProcessEvent` and `ActivityDump.Insert` functions to make sure that events are only suppressed if they are going to be added to the activity dump afterwards (as the probe could still receive kernel samples while the activity dump has already been stopped).